unison-core/src/Unison/Reference.hs

```diff
@@ -190,3 +190,10 @@ instance Show Reference where show = SH.toString . SH.take 5 . toShortHash
 instance Hashable.Hashable Reference where
   tokens (Builtin txt) = [Hashable.Tag 0, Hashable.Text txt]
   tokens (DerivedId (Id h i n)) = [Hashable.Tag 1, Hashable.Bytes (H.toBytes h), Hashable.Nat i, Hashable.Nat n]
+
+-- | A simpler version of `Id` that we'll use for some derived instances.
+data Id' = Id' H.Hash Pos deriving (Eq, Ord)
+toId' :: Id -> Id'
+toId' (Id h p _) = Id' h p
+instance Eq Id where x == y = toId' x == toId' y
+instance Ord Id where x `compare` y = toId' x `compare` toId' y
```

Comment:
I'd just write the instance out rather than introducing yet another thing ending in `'` which is only used for this one thing. :)

```suggestion
instance Eq Id where x == y = compare x y == EQ
instance Ord Id where (Id h i _) `compare` (Id h2 i2 _)  = compare h h2 <> compare i i2
```

Also happens to be fewer lines. :)

FYI, the semigroup for `Ordering` does what you'd expect:

```Haskell
instance Semigroup Ordering where
    LT <> _ = LT
    EQ <> y = y
    GT <> _ = GT
```

Reply:
Okay, cool thanks.